addons/hr_expense/models/hr_expense.py

```diff
@@ -106,47 +121,32 @@ def write(self, vals):
         return res
 
     @api.multi
-    def unlink(self):
-        if any(expense.state not in ['draft', 'cancel'] for expense in self):
-            raise UserError(_('You can only delete draft or refused expenses!'))
-        return super(HrExpense, self).unlink()
+    def view_sheet(self):
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'res_model': 'hr.expense.sheet',
+            'target': 'current',
+            'res_id': self.sheet_id.id
+        }
 
     @api.multi
     def submit_expenses(self):
         if any(expense.state != 'draft' for expense in self):
-            raise UserError(_("You can only submit draft expenses!"))
-        self.write({'state': 'submit'})
-
-    @api.multi
-    def approve_expenses(self):
-        self.write({'state': 'approve'})
-
-    @api.multi
-    def refuse_expenses(self, reason):
-        self.write({'state': 'cancel'})
-        if self.employee_id.user_id:
-            body = (_("Your Expense %s has been refused.<br/><ul class=o_timeline_tracking_value_list><li>Reason<span> : </span><span class=o_timeline_tracking_value>%s</span></li></ul>") % (self.name, reason))
-            self.message_post(body=body, partner_ids=[self.employee_id.user_id.partner_id.id])
+            raise UserError(_("You cannot report twice the same line!"))
```

Comment:
If an expense line is reported, done, or whatever, we prevent the user to report it in another sheet. Because there is the button "Submit to Manager" under the menu Action